libdnf5/repo/repo_sack.cpp

```diff
@@ -201,6 +201,7 @@ std::map<std::string, libdnf5::rpm::Package> RepoSack::add_cmdline_packages(
     }
 
     auto cmdline_repo = get_cmdline_repo();
+    cmdline_repo->load();
```

Comment:
The patch is OK, but it calls the load() only for one reason - it sets `p_impl->loaded = true;`. Everything else is redundant or skipped.

Reply:
What about to add `p_impl->loaded = true;` to `rpm::Package Repo::add_rpm_package(const std::string & path, bool with_hdrid)` or create a private method that will set repo as loaded. Or at least add comment above the line and to load() method. The description of the `load()` method does not cover the usage here => it is confusing why it is used here because it does not load libsolv repo, because it is already created.

Reply:
Yep, that's correct. The other things inside the current `load()` implementation are eventually called via the subsequent `add_rpm_package` calls. I considered simply setting the `loaded` flag to true, but the `load()` function exists there to load the repository. And that's at least strange that the `load()` method isn't called at all for the command-line repository. This could be problematic because, in the future, more functionality might be added to the `load()` method. But of course, if you have a better idea on how to address this issue, I'm open to hearing it.

Reply:
"Oh, I started writing before I saw your last message :slightly_smiling_face:

Reply:
> The description of the load() method does not cover the usage here => it is confusing why it is used here because it does not load libsolv repo, because it is already created.

Hmm, so I was debugging the existing code and after the `auto cmdline_repo = get_cmdline_repo();` statement is executed, the `solv_repo` for the `cmdline_repo` is actually nullptr yet.

And also the current docstring for the `load()` method starts with `/// Loads the repository objects into sacks.` and that's actually pretty matching.

Reply:
But when looking into the former implementation, like the checking for `solv_repo != nullptr`, it means that the `loaded` flag should have now the same semantics. So wouldn't it be better in the end to set the `loaded` flag just in the `make_solv_repo` method?